pkg/util/spanlogger/spanlogger.go

```diff
@@ -77,6 +77,31 @@ func FromContextWithFallback(ctx context.Context, fallback log.Logger) *SpanLogg
 // also puts the on the spans.
 func (s *SpanLogger) Log(kvps ...interface{}) error {
 	s.Logger.Log(kvps...)
+
+	var logAsError = false
+	errorIndex := -1
+	for i := 0; i < len(kvps)-1; i += 2 {
+		// Find out whether to log as error
+		if kvps[i] == level.Key() {
+			logAsError = kvps[i+1] == level.ErrorValue()
+			if !logAsError {
+				break
+			}
+			ext.Error.Set(s.Span, true)
+		} else if errorIndex == -1 {
+			// Check if this is the error we want to log
+			if _, ok := kvps[i+1].(error); ok && (kvps[i] == "err" || kvps[i] == "error") {
+				errorIndex = i
+			}
+		}
+		if logAsError && errorIndex != -1 {
+			s.Span.LogFields(otlog.Error(kvps[i+1].(error)))
```

Comment:
I think you need to use `errorIndex` instead of `i`, otherwise you're assuming that the level is always before the error in `kvps` ordering.